payments/models.py

```diff
@@ -39,6 +40,63 @@ def __setattr__(self, key, value):
         return None
 
 
+class BaseSubscription(models.Model):
+    token = models.CharField(
+        _("subscription token/id"),
+        help_text=_("Token/id used to identify subscription by provider"),
+        max_length=255,
+        default=None,
+        null=True,
+        blank=True,
+    )
+    payment_provider = models.CharField(
+        _("payment provider"),
+        help_text=_("Provider variant, that will be used for payment renewal"),
+        max_length=255,
+        default=None,
+        null=True,
+        blank=True,
+    )
+    subscribtion_data = models.JSONField(
+        _("subscription data"),
+        help_text=_("Provider-specific data associated with subscription"),
+        default=dict,
+    )
+
+    class TimeUnit(enum.Enum):
+        year = "year"
+        month = "month"
+        day = "day"
+
+    def set_recurrence(self, token: str, **kwargs):
+        """
+        Sets token and other values associated with subscription recurrence
+        Kwargs can contain provider-specific values
+        """
+        self.token = token
+        self.subscribtion_data = kwargs
+
+    def get_period(self) -> int:
+        raise NotImplementedError
+
+    def get_unit(self) -> TimeUnit:
+        raise NotImplementedError
+
+    def cancel(self):
+        """
+        Cancel the subscription by provider
+        Used by providers, that use provider initiated subscription workflow
+        Implementer is responsible for cancelling the subscription model
```

Comment:
Could you please clarify this docstring a bit? I am not sure if it really says what it is supposed to do. I mean, what do you mean by:

1. Cancel the subscription **by provider** - is there any "by <something>" needed? Isn't "cancel the subscription" enough?
2. **Used by providers**, that use provider initiated subscription workflow - cannot it potentially be used by anyone, if the provider supports it?
3. **Implementer is responsible for cancelling the subscription model**